crates/generator/src/generator.rs

```diff
@@ -125,42 +122,64 @@ impl Generator {
         &self.account_lock_manage
     }
 
-    fn build_machine<'a, S: State + CodeStore, C: ChainStore>(
+    #[allow(clippy::too_many_arguments)]
+    fn machine_run<'a, S: State + CodeStore, C: ChainStore>(
         &'a self,
-        run_result: &'a mut RunResult,
         chain: &'a C,
         state: &'a S,
         block_info: &'a BlockInfo,
         raw_tx: &'a RawL2Transaction,
         max_cycles: u64,
-    ) -> Machine<'a> {
-        let global_vm_version = smol::block_on(async { *GLOBAL_VM_VERSION.lock().await });
-        let vm_version = match global_vm_version {
-            0 => VMVersion::V0,
-            1 => VMVersion::V1,
-            ver => panic!("Unsupport VMVersion: {}", ver),
-        };
-        let core_machine = vm_version.init_core_machine(max_cycles);
-        let machine_builder = DefaultMachineBuilder::new(core_machine)
-            .syscall(Box::new(L2Syscalls {
-                chain,
-                state,
-                block_info,
-                raw_tx,
-                rollup_context: &self.rollup_context,
-                account_lock_manage: &self.account_lock_manage,
-                result: run_result,
-                code_store: state,
-            }))
-            .instruction_cycle_func(Box::new(instruction_cycles));
-        let default_machine = machine_builder.build();
-
-        #[cfg(has_asm)]
-        let machine = AsmMachine::new(default_machine, None);
-        #[cfg(not(has_asm))]
-        let machine = TraceMachine::new(default_machine);
-
-        machine
+        code_bytes: Bytes,
+        aot_code_opt: Option<&AotCode>,
+    ) -> Result<RunResult, TransactionError> {
+        let mut run_result = RunResult::default();
+        let used_cycles;
+        let exit_code;
+
+        {
+            let t = Instant::now();
+            let global_vm_version = smol::block_on(async { *GLOBAL_VM_VERSION.lock().await });
```

Comment:
We should cache the compiling vm version, and use that version. In the current implementation we may use VM::V1 to run AOT code compiling under VM::V0 argument.